aws/src/main/java/org/apache/iceberg/aws/AwsClientFactories.java

```diff
@@ -97,6 +97,7 @@ static class DefaultAwsClientFactory implements AwsClientFactory {
     private Boolean s3AccelerationEnabled;
     private String dynamoDbEndpoint;
     private String httpClientType;
+    private Boolean s3DualStackEnabled;
```

Comment:
[minor] is box type required here ?

Reply:
Thank you for your review and suggestions!

Here is what I was thinking when writing `line 100`: 
1. The method `dualStackEnabled(Boolean dualstackEndpointEnabled)` specified its argument type as a boxed boolean. The doc I refer to: 
https://sdk.amazonaws.com/java/api/latest/software/amazon/awssdk/awscore/client/builder/AwsClientBuilder.html
2. Other similar flags like `s3AccelerationEnabled`, `s3UseArnRegionEnabled`, and `s3PathStyleAccess` use boxed boolean.

Hence, I think it may be a good idea to keep this pattern and use `Boolean` as the type for `s3DualStackEnabled`.

I also tried the primitive type version of `s3DualStackEnabled`. It turns out that the Autoboxing works well and the code compiles and passes `iceberg-aws` test. So it seems the boxed type here is not a strict requirement